ThermalTalk/Common/BasePrinter.cs

```diff
@@ -350,41 +469,29 @@ public void Dispose()
         /// <summary>
         /// Close serial connection
         /// </summary>
-        /// <param name="diposing">True to close connection</param>
-        protected virtual void Dispose(bool diposing)
+        /// <param name="disposing">True to close connection</param>
+        protected virtual void Dispose(bool disposing)
         {
             if (Connection != null)
             {
                 Connection.Dispose();
             }
+            
+            _docBuffer.Dispose();
         }
 
         #region Protected
         /// <summary>
-        /// Send payload over serial port. If port
-        /// is not open, this will open the port before writing.
-        /// The port will be closed when the write completes or fails.
+        /// Append print command data to the end of the current document.
+        /// Document is not actually printed until <see cref="FormFeed"/> is called.
         /// </summary>
         /// <param name="payload"></param>
-        protected void internalSend(byte[] payload)
+        /// <returns>ReturnCode.Success if successful, ReturnCode.UnsupportedCommand if payload.Length == 0,
+        /// and ReturnCode.ExecutionFailure otherwise.</returns>
+        protected ReturnCode AppendToDocBuffer(byte[] payload)
         {
-            // Do not send empty packets
-            if (payload.Length == 0)
-            {
-                return;
-            }
-
-            try
-            {
-                Connection.Open();
-
-                Connection.Write(payload);
-            }
-            catch { /* Do nothing */ }
-            finally
-            {
-                Connection.Close();
-            }
+            _docBuffer.Write(payload);
```

Comment:
If payload is null (and it can be if a user passes in null to `SendRaw(byte[] raw)`) this will throw an exception. Do we want to try/catch here?

Reply:
Sorry I had the wrong base target on this PR. Good catch, I'll check for that and see if that's still a problem.